### PR TITLE
ISPN ActionsService support

### PR DESCRIPTION
**Pull request by jshaughn**
@lucasponce Please take a look.  I removed the unused executor from ispnalertsserviceimpl, we can put it back if you think we need it.  I also removed use of an executor in ispnactionsserviceimpl.  My understanding is the executor was needed only for c* concurrent writes.

Also, I wasted a lot of time trying to use the "in" operator in an ispn query.  I don't see that I was doing anything wrong but it seems not to work.  Instead I'm using an exploded "or" approach.

### CODE REVIEW

**Comment by lucasponce** [comment]
First comment is the executor is needed.
It helps to not block the engine when executing actions.
I will take a close look this morning with more comments.

File: engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java [hunk 1]
```diff
@@ -61,32 +95,312 @@ public void setActionsCacheManager(ActionsCacheManager actionsCacheManager) {
...
+                query.append(inClause("result", results));
+            }
+        }
+
+        List<IspnAction> ispnActions = queryFactory.create(query.toString()).list();
+        return prepareActionsPage(ispnActions.stream().map(a -> a.getAction()).collect(Collectors.toList()), pager);
+    }
+
+    // An exploded "in" clause because the actual one seems not to work
+    private String inClause(String field, Set<String> values) {
```
Comment (lucasponce) [inline engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java]: I use the same pattern in alerts criteria.
I think you need an iterator here or you will have a " and ( or field = 'value' or field = 'value')" and this problably will throw a syntax error.
I used a 
```
Iterator<String> iter = values.iterator();
while (iter.hasNext()) {
      String value = iter.next();
      sb.append("field = '").append(value).append("'");
      if (iter.hasNext()) {
        sb.append(" or ");
      }
}
```

File: engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java [hunk 2]
```diff
@@ -61,32 +95,312 @@ public void setActionsCacheManager(ActionsCacheManager actionsCacheManager) {
         this.actionsCacheManager = actionsCacheManager;
     }
 
-    public void setExecutor(ExecutorService executor) {
```
Comment (lucasponce) [inline engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java]: The executor has the meaning to not block the engine thread while writting in the backend or in actions, when processing actions.
So I would leave it.
Once that we start the round of itest on Ispn backend we can decide if we refactor it but I wouldn't introduce this change now.

**Comment by lucasponce** [comment]
Merging, any comments can be fixed in upcoming PRs.

File: engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java (hunk 1 above)
Comment (jshaughn) [inline engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java]: > I think you need an iterator here

No, this is OK as is.  Your approach is good, too.

**Comment by jshaughn** [comment]
> First comment is the executor is needed

OK, we can add some executor use back in as needed.

File: engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java (hunk 1 above)
Comment (lucasponce) [inline engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnActionsServiceImpl.java]: true, the separator was initially empty. I didn't see it :-).

**Pull request merged**